pxr/base/tf/stringUtils.cpp

```diff
@@ -806,8 +806,8 @@ TfDictionaryLessThan::_LessImpl(const string& lstr, const string& rstr) const
             // Add 5 mod 32 makes '_' sort before all letters.
             return ((l + 5) & 31) < ((r + 5) & 31);
         }
-        else if (IsDigit(l) | IsDigit(r)) {
-            if (IsDigit(l) & IsDigit(r)) {
+        else if (IsDigit(l) || IsDigit(r)) {
+            if (IsDigit(l) && IsDigit(r)) {
                 // We backtrack to find the start of each digit string, then we
```

Comment:
The use of bitwise operators here are intentional, to make the generated code branchless.

Reply:
Ah, sorry.  I'll back that out.

Reply:
But then, further down you've got:

```
else if (!IsAlpha(l) || !IsAlpha(r)) {
```

Reply:
I was checking this with a simple test function in compiler explorer and didn't see any logical difference in the generated output with `-O3`.  But I might have missed something.